### PR TITLE
Add G102 Unit Tests

### PR DESCRIPTION
**Pull request by duytnguyendtn**
This PR adds some test G102 data and adapts the existing G141 unit tests for G102. ~There's a lot of overlap code, so I'm thinking of genericizing the tests out a bit to minimize the Copy and Paste~ Complete!

EDIT: This PR has also parametrized out the tests into two separate tests describing the proper use case: image2world and image2grism. Now we run those identical tests, just swapping out the FLT file we use for those tests.

### CODE REVIEW

File: astrogrism/tests/test_wfc3_g141_grizli.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from astrogrism.tests.test_image2grism import _image2grism
```
[the remark this answers is not in the document]
Comment (duytnguyendtn) [reply astrogrism/tests/test_wfc3_g141_grizli.py]: Got confirmation from PO:

> We can assume that GRISMCONF and Grizli are consistently giving similar results. I believe GRISMCONF was written later than Grizli so GRISMCONF must have done sanity check with Grizli to make sure that results are as expected. Our goal is not to compare GRISMCONF and Grizli but one of those vs Astrogrism so checking Astrogrism with GRISMCONF should be fine.

File: astrogrism/tests/test_image2grism.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    x_center = 500
+    y_center = 500
+    dt = np.abs(1/(C.DISPX('+1',
+                           x_center,
+                           y_center,
+                           1) - C.DISPX('+1',
+                                        x_center,
+                                        y_center,
+                                        0)))
+    t = np.arange(0, 1, dt)
```
[the remark this answers is not in the document]
Comment (duytnguyendtn) [reply astrogrism/tests/test_image2grism.py]: I have no objections!

File: astrogrism/tests/test_image2grism.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+from astrogrism import GrismObs
+from astropy.utils.data import download_file
+import grismconf
+import numpy as np
+import pytest
+
+# TODO: Switch to importlib
+test_dir = pathlib.Path(__file__).parent.absolute()
+testdata = [
+    ('G141', download_file('https://github.com/npirzkal/aXe_WFC3_Cookbook/raw/main/cookbook_data/G141/ib6o23rsq_flt.fits')), # noqa
```
[the remark this answers is not in the document]
Comment (duytnguyendtn) [reply astrogrism/tests/test_image2grism.py]: I copied in the file, but didn't excise the data. If we start thinking about optimizations for the package, then I think we can revisit at that time

**Comment by duytnguyendtn** [comment]
@rosteen I added roundtripping for the i2w and i2g tests

https://github.com/spacetelescope/astrogrism/blob/0696850542311c01596694cf986d0da00df6eed9/astrogrism/tests/test_image2world.py#L47-L52
and
https://github.com/spacetelescope/astrogrism/blob/0696850542311c01596694cf986d0da00df6eed9/astrogrism/tests/test_image2grism.py#L68-L74
respectively. See if you think this has enough coverage?

**Comment by rosteen** [comment]
Looks good to me, thanks for addressing all the comments! There are some `ResourceWarning` warnings about unclosed files, but I address at least some of those in #7 so I think we can reassess if there are more changes needed there.

**Pull request merged**